app/common/update.py

```diff
@@ -253,3 +258,65 @@ def merge_local_db():
 
         logger.info(str(records_inserted) + " records were inserted into local db.")
         logger.info(str(records_updated) + " records in local db were updated.")
+        
+
+def download_dat_files():
+    config = load_user_config()
+    valid_directory = False
+    
+    if not config["config"]["installdirectory"]:
+    
+        #First, let's check the default location and see if it's installed there
+        path = 'C:/Program Files (x86)/SquareEnix/DRAGON QUEST X/Game/Content/Data'
+        
+        if os.path.exists(path):
+            update_user_config('config', 'installdirectory', path)
+        else:
+            #Doesn't exist. Let's prompt the user for it.
+            warning_message(
+            title="[dqxclarity] Couldn't Find Directory",
+            message="Could not find DQX directory. Please select\n\nthe Data folder located in your\n\nDQX install location."
+        )
+            
+            path = askdirectory()
+            
+            #Path looks legit.
+            if 'Game/Content/Data' in path:
+                update_user_config('config', 'installdirectory', path)
+                valid_directory = True
+            else:
+                message_box_fatal_error(
+                    title="[dqxclarity] Invalid Directory",
+                    message="The path you provided is not a valid DQX path.\n\nPlease try again."
+                )
+    else:
+        if 'Game/Content/Data' in config["config"]["installdirectory"]:

```

Comment:
We should validate the path listed in the install directory here -- instead of assuming it's ok just based on the subpath (and handle it if it's wrong).